### ISSUE TITLE
Start position for loops and actual playtime

### ISSUE THREAD

**Issue by mromanov24** [spatialaudio/python-sounddevice#26]
Hello there,

i work on a MUSHRA like listening test for multispeakersetups at the moment and use `sounddevice.play()` and `stop()` to play multichannel audio files out of my GUI. I can switch between sounds and start them playing in a loop. 

But what i am not able to do is to let the the loop start from a specific start point. What i would like to do is to switch between the sounds without restarting them. So get the actual time position of the track i just stopped and feed that information as the starting point for an other track so that i can compare the stimuli better.

Is there a possibility to do that or a known workaround to achieve this behaviour? Someone experience with that?

Thanks and regards,

Michael

**Comment by mgeier** [comment]
Welcome to Github!

Are you planning to release your MUSHRA GUI as open source software?
I think this would be great!

There is no way to specify an offset when using `sounddevice.play()` *and* still play the whole array in a loop.

The idea of the `play()` function is to provide a very quick and easy way to play a NumPy array during interactive exploration. Whenever you want more control over the playback, you should implement it on your own using the callback API.

A specific disadvantage of the `play()` function in your context is that it each time opens and closes a PortAudio stream. Therefore, gapless playback of consecutive calls is not guaranteed. Actually, this is never *guaranteed*, but in this case there will be no indication in case of a gap in the playback. Also, depending on your audio hardware/driver, there might be an audible click when starting and stopping a stream. This typically happens with cheap and/or on-board sound cards, but for your listening tests you'll hopefully use a proper sound card anyway.

In your case, I would highly recommend that you open the `OutputStream` in the beginning of the test run (ideally in a `with` statement), let all the user interaction happen while the stream is open, and close the stream only at the end.
This way you will know if playback was gapless.
In addition, you should probably consider making a crossfade when switching between stimuli.

**Comment by mgeier** [comment]
@mromanov24 Any news?

**Comment by mromanov24** [comment]
Hello,

sorry for the long time off. Been on travels. 

I used the script to control an other system over osc for my last project because i needed a quick solution but will work on the player in the next month to get an accurate internal playback.

The initial code of the the listening test can be found here: https://github.com/maxgiller/listening-test
Its a work of a friend of mine. In his initial version it controls reaper. 

I optimized this to a flexible standalone tool for different kinds of listening tests and some other functionality.
As soon as i cleaned everything up i will commit an update to Maxs initial commit.

Regards

MR
